### ISSUE TITLE
Provide FromIterator implementation

### ISSUE THREAD

**Issue by CGMossa** [lumol-org/soa-derive#22]
Lovely package! Here's my usecase though, set in terms of the readme example: I have `Vec<Cheese>` and I attempted to do:

```rust
let cheeses: Vec<Cheese> = ...;
let CheeseBoard: CheeseVec = cheeses.into_iter().collect();
```

My example is similar to this, and I am getting this:

```rust
error[E0277]: a value of type `boar_parameters::MortalityProbabilityVec` cannot be built from an iterator over elements of type `boar_parameters::MortalityProbability`
   --> src\boar_parameters.rs:374:14
    |
374 |             .collect();
    |              ^^^^^^^ value of type `boar_parameters::MortalityProbabilityVec` cannot be built from `std::iter::Iterator<Item=boar_parameters::MortalityProbability>`
    |
    = help: the trait `std::iter::FromIterator<boar_parameters::MortalityProbability>` is not implemented for `boar_parameters::MortalityProbabilityVec`
```

So basically, there should be a `FromIterator` for this?

**Comment by Luthaf** [comment]
You are right, there should be an implementation of `FromIterator<Item=Cheese> for CheeseVec` or something similar. Do you think you could send a PR adding it?

In the mean time, you can do
```rust
let cheeses: Vec<Cheese> = ...;
let mut cheese_board = CheeseVec::new();
for cheese in cheeses {
    cheese_board.push(cheese);
}
```

**Comment by CGMossa** [comment]
I honestly do not think I can send a pr adding it. I'm not even remotely good enough to do that, but I'll provide whatever else support I can, if a pr emerges.

I implemented one by hand and it worked wonderful. That's why I sort of know it is possible to do.

**Comment by Luthaf** [comment]
Fixed in #23
